### PR TITLE
Update QhullUser.cpp

### PR DESCRIPTION
**Pull request by stephanmg**
Getting rid of uintptr_t to make the code C++98 compatible.

### CODE REVIEW

**Comment by SpaceIm** [comment]
This modification fails on Visual Studio with `QhullUser.cpp(101,170): error C2440: '<function-style-cast>': cannot convert from 'initializer list' to 'orgQhull::QhullError'`. It very likely fails with all compilers.

File: src/libqhullcpp/QhullUser.cpp [hunk 1]
```diff
@@ -98,7 +98,7 @@ captureOff()
         throw QhullError(10080, "Qhull error: QhullUser::captureOn not call before QhullUser::captureOff for QhullUser 0x%llx", 0, 0, 0.0, this);
     }
     if(qh()->cpp_user!=this){
-        throw QhullError(10081, "Qhull error: conflicting QhullUser (0x%llx) for QhullUser::captureOff().  Does not match 'this' (0x...%X)", int((uintptr_t)this), 0, 0.0, qh()->cpp_user);
+        throw QhullError(10081, "Qhull error: conflicting QhullUser (0x%llx) for QhullUser::captureOff().  Does not match 'this' (0x...%X)", this, 0, 0.0, qh()->cpp_user);
```
Comment (SpaceIm) [inline src/libqhullcpp/QhullUser.cpp:101]:
```suggestion
        throw QhullError(10081, "Qhull error: conflicting QhullUser (0x%llx) for QhullUser::captureOff().  Does not match 'this' (0x...%X)", *(int *)this, 0, 0.0, qh()->cpp_user);
```

**Comment by SpaceIm** [comment]
There is also C++11 in QhullUser.h

Fix:

```patch
--- a/src/libqhullcpp/QhullUser.h
+++ b/src/libqhullcpp/QhullUser.h
@@ -36,8 +36,8 @@ namespace orgQhull{
 #//!\name Fields
         QhullQh *       qh_qh;          //!< QhullQh/qhT for access to libqhull_r
         void *          previous_user;  //!< previous qh.cpp_user, restored on deconstruction
-        std::vector< std::vector< double>> doubles_vector;  //! vectors for capturing ints and doubles
-        std::vector< std::vector< int>> ints_vector;
+        std::vector< std::vector< double> > doubles_vector;  //! vectors for capturing ints and doubles
+        std::vector< std::vector< int> > ints_vector;
         std::vector< int>               fprintf_ints;
         std::vector< double>            fprintf_doubles;
         std::vector< int>               fprintf_codes;
@@ -77,11 +77,11 @@ namespace orgQhull{
         void            clearIntsVector() { ints_vector.clear(); }
         const std::vector< int> &codes() const { return fprintf_codes; }
         int             delaunayDim() const { return delaunay_dim; }
-        const std::vector< std::vector< double>> &doublesVector() const { return doubles_vector; }
+        const std::vector< std::vector< double> > &doublesVector() const { return doubles_vector; }
         const std::vector< double> &doubles() const { return fprintf_doubles; }
         int             firstCode() const { return (fprintf_codes.size() == 0 ? -1 : fprintf_codes[0]); }
         const std::vector< int> &ints() const { return fprintf_ints; }
-        const std::vector< std::vector< int>> &intsVector() const { return ints_vector; }
+        const std::vector< std::vector< int> > &intsVector() const { return ints_vector; }
         int             numDoubles() const { return (int)doubles_vector.size();  }
         int             numFacets() const { return num_facets; }
         int             numInts() const { return (int)ints_vector.size(); }
```

and also range for loop to replace in `src/user_eg3/user_eg3_r.cpp`

**Comment by cbbarber** [comment]
The third argument to QhullError should be the low order bits of 'this'  to aid in debugging the reported error.    '__intptr_t' would allow the cast to int.    It was suggested in #65
        note: suggested alternative: ‘__intptr_t’

Thanks for fixing the other C++11 issues.

**Comment by cbbarber** [comment]
Fixed in Qhull 8.0.1.   Many thanks for reporting these problems.

**Pull request closed without merging**